### DISCUSSION TITLE
Always upload build artefacts to the pull request

### DISCUSSION THREAD

**Discussion by tino-teuber** [microsoft/AL-Go#1055] [Q&A]
Hello!

Is there a way that the build artefacts (app, test app, dependencies) are always attached to the pull request? 

When I look at the pull request handler workflow, it seems that the build artefacts are not attached due to the following condition:

https://github.com/microsoft/AL-Go/blob/ff6ed5761ab14cd90142fb9813f997cc5e16eef1/Templates/AppSource%20App/.github/workflows/PullRequestHandler.yaml#L103

Can I perhaps use a setting to ensure that the build artefacts are always uploaded to the pull request?

Thank you very much! 🙏

**Comment by freddydk** [comment]
Right now, there is no such setting.
What would the purpose be of storing build artifacts from PRs?
How/why would you use them?

**Reply by tino-teuber** [reply]
The build artifacts (the BC apps) are required by our consultants to test the functionality. Only when the code review (Developer) and functional test (Consultant) are positive are the changes merged into the main branch.

We are currently stuck with AL-Go version 2.4 because - according to my information - it is the last version that uploads build artifacts in the pull request.

**Comment by freddydk** [comment]
Will investigate what we can do

**Comment by freddydk** [comment]
Even if we create artifacts on the PR - they would need to manually deploy those right?
QQ - why wouldn't you have a feature branch on which CI/CD would push the feature to a QA environment, which could be tested by the reviewer?

**Reply by tino-teuber** [reply]
Yes, that's right. Our consultants still have to publish the artifacts manually (install the BC apps in the BC Extension Management)

Our consultants - who do not have access to GitHub - decide for themselves when to test which feature. They test each feature individually in their QR environment (BC Cloud Sandbox).

**Reply by tino-teuber** [reply]
I have just discovered the pull request [[DRAFT] Add support for indirect template + customisations](https://github.com/microsoft/AL-Go/pull/853). Couldn't I also implement the upload of build artifacts with the ‘Custom Steps’ and ‘Custom Jobs’ features implemented in the pull request?

**Comment by freddydk** [comment]
In the latest version of AL-Go for GitHub we have implemented Publish To Environment from Pull Request - and artifacts are now always published on PRs.
